apps/meteor/app/lib/server/lib/notifyListenerOnRoomChanges.ts

```diff
@@ -0,0 +1,27 @@
+import { api, dbWatchersDisabled } from '@rocket.chat/core-services';
+import type { IRoom } from '@rocket.chat/core-typings';
+import { Rooms } from '@rocket.chat/models';
+
+type ClientAction = 'inserted' | 'updated' | 'removed';
+
+export async function notifyListenerOnRoomChanges(
+	rid: IRoom['_id'],
+	clientAction: ClientAction = 'updated',
+	existingRoomData?: IRoom,
+	args?: { [key: string]: any },
+): Promise<void> {
+	if (!dbWatchersDisabled) return;
+
+	const room = existingRoomData || (await Rooms.findOneById(rid));
+
+	if (room) {
+		void api.broadcast('watch.rooms', { clientAction, room: { ...room, ...args } });
+	}
+}
+
+export async function notifyListenerOnRoomsChanges(rooms: IRoom[], clientAction: ClientAction = 'updated'): Promise<void> {
+	if (!dbWatchersDisabled) return;
+	if (!rooms.length) return;
```

Comment:
maybe this is personal opinion, but I prefer doing this way, makes it easier to read/spot:
```suggestion
	if (!dbWatchersDisabled) {
		return;
	}
	if (!rooms.length) {
		return;
	}
```

we may need a proper code style definition for this one =)